### PR TITLE
Cigarette lighting QoL

### PR DESCRIPTION
**Pull request by St0rmC4st3r**
## About The Pull Request

Adds the ability to pull out a lighter from a cig pack on Ctrl+Click.

## Why It's Good For The Game

QoL. Makes smokers actually smoke and get more cigs, as simply lighting a cigarette is no longer a chore.

## Testing Photographs and Procedure

<details>

image

</details>

## Changelog
:cl:
add: You can now pull out a lighter from a cigarette pack by using Ctrl+Click.
/:cl:

<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. --

### CODE REVIEW

**Comment by Marshmellow105** [comment]
I was just thinking about this earlier today, awesome change.

**Review by itsmeow — APPROVED** [review]

File: code/game/objects/items/storage/fancy.dm [hunk 1]
```diff
@@ -146,6 +146,9 @@
 /obj/item/storage/fancy/cigarettes/examine(mob/user)
 	. = ..()
 	. += "<span class='notice'>Alt-click to extract contents.</span>"
+	var/obj/item/lighter/L = locate(/obj/item/lighter) in contents
+	if(L && contents.len > 0)
```
Comment (ike709) [inline code/game/objects/items/storage/fancy.dm]:
```suggestion
	if(L)
```
The second condition is always true if the first condition is true.

File: code/game/objects/items/storage/fancy.dm [hunk 2]
```diff
@@ -159,6 +162,18 @@
 	else
 		to_chat(user, "<span class='notice'>There is nothing left in the pack.</span>")
 
+/obj/item/storage/fancy/cigarettes/CtrlClick(mob/living/carbon/user)
+	if(!istype(user) || !user.canUseTopic(src, BE_CLOSE, ismonkey(user)))
+		return
+	var/obj/item/I = locate(/obj/item/lighter) in contents
+	if(I && contents.len > 0)
```
Comment (ike709) [inline code/game/objects/items/storage/fancy.dm]:
```suggestion
	if(I)
```
ditto

File: code/game/objects/items/storage/fancy.dm [hunk 3]
```diff
@@ -159,6 +162,18 @@
...
+		SEND_SIGNAL(src, COMSIG_TRY_STORAGE_TAKE, I, user)
+		user.put_in_hands(I)
+		contents -= I
+		to_chat(user, "<span class='notice'>You take \a [I] out of the pack.</span>")
+	else
+		to_chat(user, "<span class='notice'>There is no lighter in the pack.</span>")
```
Comment (ike709) [inline code/game/objects/items/storage/fancy.dm]:
```suggestion
		to_chat(user, "<span class='warning'>There is no lighter in the pack.</span>")
```
We're not super consistent about this but IMO successes should be notices and failures should be warnings.

**Review by ike709 — CHANGES_REQUESTED** [review]
.

File: code/game/objects/items/storage/fancy.dm (hunk 1 above)
Comment (itsmeow) [reply code/game/objects/items/storage/fancy.dm]: Empty lists are truthy afaik? either way just use length()

Comment (itsmeow) [reply code/game/objects/items/storage/fancy.dm]: Yea empty lists are truthy

Comment (itsmeow) [reply code/game/objects/items/storage/fancy.dm]: nvm i thought L and contents were the same list if(L) is good

**Review by ike709 — APPROVED** [review]

**Pull request merged**